### ISSUE TITLE
how to check the status for promise object

### ISSUE THREAD

**Issue by wangvisual** [FGasper/p5-Promise-XS#8]
I'm try to move my project to migrate from Promises to Promise::XS as I used a lot of promises thus the performance is very important.

And now I have 2 issues need to check:
1. Can Promise::XS support EV loop, I load AnyEvent in the project but the loop is from EV(EV::run). Short test shows that it can work, don't know if there any issues.
2. I implemented an async queue that handle incoming requests, but sometimes I need to know if a promise(not deferred) is resolved or not, is there any API to check the status of a promise? https://metacpan.org/pod/Promises::Promise has APIs for the status.

Thanks.

**Comment by wangvisual** [comment]
EV loop is fine, as AnyEvent is an wrapper over EV for when using epool.
Still it's would be better to have status APIs for Promise.

**Comment by FGasper** [comment]
> Can Promise::XS support EV loop

Yes, via AnyEvent, IO::Async, or Mojo. Feel free to contribute a direct adaptor; I’m unlikely to create one myself.

> is there any API to check the status of a promise?

No. You can achieve this pretty simply via the existing interface, though:
```
my $is_done;
$promise->finally( sub { $is_done = 1 } )->then( … )
```
Given that ES6 promises lack a facility to check the promise’s status directly, is there a reason why Perl promises need such an interface?
